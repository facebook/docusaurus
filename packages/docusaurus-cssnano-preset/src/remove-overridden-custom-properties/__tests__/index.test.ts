/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import vfile from 'to-vfile';
import postcss from 'postcss';
import postCssRemoveOverriddenCustomProperties from '../index';

const processFixture = async (name: string) => {
  const input = await vfile.read(
    path.join(__dirname, '__fixtures__', `${name}.css`),
    'utf8',
  );
  const output = postcss([postCssRemoveOverriddenCustomProperties]).process(
    input,
  );

  return output.css;
};

describe('remove-overridden-custom-properties', () => {
  it('overridden custom properties should be removed', async () => {
    await expect(processFixture('normal')).resolves.toMatchSnapshot();
  });

  it('overridden custom properties with `!important` rule should not be removed', async () => {
    await expect(processFixture('important_rule')).resolves.toMatchSnapshot();
  });
});
