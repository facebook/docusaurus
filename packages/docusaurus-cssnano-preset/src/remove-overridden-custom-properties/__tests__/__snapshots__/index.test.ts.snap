// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remove-overridden-custom-properties overridden custom properties should be removed 1`] = `
"/* stylelint-disable docusaurus/copyright-header, declaration-block-no-duplicate-custom-properties */
:root {
  --color-secondary: green;
  --color-primary: blue;
  --color-header: gray;
}

.non-root {
  --color-primary: red;
  --color-primary: red;
}
"
`;

exports[`remove-overridden-custom-properties overridden custom properties with \`!important\` rule should not be removed 1`] = `
"/* stylelint-disable docusaurus/copyright-header, declaration-block-no-duplicate-custom-properties */
:root {
  --color-primary: blue;
  --color-header: gray !important;
  --color-secondary: yellow !important;
}
"
`;
